src/checkout-buttons/strategies/masterpass-button-strategy.ts

```diff
@@ -0,0 +1,149 @@
+import { CheckoutButtonInitializeOptions, CheckoutButtonOptions } from '../';
+import { Checkout, CheckoutActionCreator, CheckoutStore, InternalCheckoutSelectors } from '../../checkout';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType
+} from '../../common/error/errors';
+import { bindDecorator as bind } from '../../common/utility';
+import { PaymentMethod } from '../../payment';
+import {
+    Masterpass,
+    MasterpassCheckoutOptions,
+    MasterpassScriptLoader
+} from '../../payment/strategies/masterpass';
+
+import { CheckoutButtonStrategy, MasterpassButtonInitializeOptions } from './';
+
+export default class MasterpassButtonStrategy extends CheckoutButtonStrategy {
+    private _masterpassClient?: Masterpass;
+    private _signInButton?: HTMLElement;
+    private _stateCheckout?: InternalCheckoutSelectors;
+    private _methodId?: string;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super();
+    }
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const { masterpass: masterpassOptions, methodId } = options;
+
+        if (!masterpassOptions || !methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "options.masterpass" argument is not provided.');
+        }
+        this._methodId = methodId;
+
+        if (this._isInitialized) {
+            return super.initialize(options);
+        }
+
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
```

Comment:
Hey @cmedinilla, sorry if I didn't explain myself clearly last time, I didn't mean to remove `loadPaymentMethod` and `loadDefaultCheckout`. You still need those. All I'm asking is not to read it from `stateCheckout` and store a separate reference of it. So with what you've done so far, the code could be something like the following:

```ts
initialize(options: CheckoutButtonInitializeOptions): Promise <void> {
    const { masterpass: masterpassOptions, methodId } = options;

    if (!masterpassOptions || !methodId) {
        throw new InvalidArgumentError('Unable to proceed because "options.masterpass" argument is not provided.');
    }

    if (this._isInitialized) {
        return super.initialize(options);
    }

    this._methodId = methodId;

    return Promise.all([
        this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId)),
        this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout()),
    ])
        .then(() => this._masterpassScriptLoader.load(this._getPaymentMethod().config.testMode))
        .then(masterpass => {
            this._masterpassClient = masterpass;
            
            this._createSignInButton(masterpassOptions);

            return super.initialize(options);
        });
}
```

Reply:
I made some changes, now i am not saving the state anymore and i am only using the data to create the payload